solc/CommandLineInterface.cpp

```diff
@@ -402,6 +402,8 @@ outputs the components specified in the options at standard output or in files i
 the output directory, if specified.
 Example: solc --bin -o /tmp/solcoutput contract.sol
 
+When - is specified as one of the input files it will add the current directory to the allowed directories and read from standard input
```

Comment:
What is the justification for adding the current directory to the allowed directories if you use stdin?
I think there is a good argument to actually *not* do it. Note that the "allowed directories" is a security feature, not a way to find input files.